File: 04-12-2024_Make_String_a_Subsequence_Using_Cyclic_Increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int i = 0, j = 0;
        while (i < str1.length() && j < str2.length()) {
            if (str1.charAt(i) == str2.charAt(j)) {
                j++;
            } else if (str1.charAt(i) == 'z') {
                if ('a' == str2.charAt(j)) {
                    j++;
                }
            } else {
                if ((char)(str1.charAt(i) + 1) == str2.charAt(j)) {
                    j++;
                }
            }
            i++;
        }
        return j == str2.length();
    }
}

File: 04-12-2024_Make_String_a_Subsequence_Using_Cyclic_Increments.py
class Solution(object):
    def canMakeSubsequence(self, str1, str2):
        i=0
        j=0
        while (i<len(str1) and j<len(str2)):
            if (str1[i]==str2[j]):
                j=j+1
            elif  (str1[i]=="z"):
                if ("a"==str2[j]):
                    j=j+1
            elif (str1[i]!="z"):
                if (chr(ord(str1[i])+1)==str2[j]):
                    j=j+1
            i=i+1
        if (j==len(str2)):
            return True
        else:
            return False
